pkgs/os-specific/linux/lsirec/default.nix

```
@@ -0,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+}:
+
+stdenv.mkDerivation rec {
+  pname = "lsirec";
+  version = "unstable-2019-03-03";
+
+  src = fetchFromGitHub {
+    owner = "marcan";
+    repo = "lsirec";
+    rev = "2dfb6dc92649feb01a3ddcfd117d4a99098084f2";
+    sha256 = "sha256-8v+KKjAJlJNpUT0poedRTQfPiDiwahrosXD35Bmh3jM=";
+  };
+
+  postPatch = ''
+    substituteInPlace sbrtool.py \
```

There's actually a really nice function in place to do just this.

```suggestion
    patchShebangs sbrtool.py
```

---

I just tried that but it doesn't work. The shebang stays at /usr/bin/python3. I even tried installing first to add the executable bit but that didn't change anything either. I'll just keep using the `substituteInPlace` function.

---

nix does not know which python it should use.

```diff
diff --git a/pkgs/os-specific/linux/lsirec/default.nix b/pkgs/os-specific/linux/lsirec/default.nix
index 2c015c30b15..1f0b5f8bc3f 100644
--- a/pkgs/os-specific/linux/lsirec/default.nix
+++ b/pkgs/os-specific/linux/lsirec/default.nix
@@ -15,9 +15,10 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-8v+KKjAJlJNpUT0poedRTQfPiDiwahrosXD35Bmh3jM=";
   };
 
+  buildInputs = [ python3 ];
+
   postPatch = ''
-    substituteInPlace sbrtool.py \
-      --replace '/usr/bin/python3' '${python3}/bin/python'
+    patchShebangs sbrtool.py
   '';
 
   installPhase = ''
```

---

Thank you. This works. I don't actually need to manually call `patchShebangs` in this case.